### DISCUSSION TITLE
Welcome to nes-starter-kit Discussions!

### DISCUSSION THREAD

**Discussion by potatolain** [igwgames/nes-starter-kit#31] [Announcements]
## Welcome!

I'm testing out discussions as a more open way to collaborate on nes-starter-kit. Feel free to: 
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and be open-minded. No hatred, no sexism, no bigotry, no homophobia/racism/transphobia/etc. Respect everyone else here and we'll be okay. (See also: [Github's community guidelines](https://docs.github.com/en/site-policy/github-terms/github-community-guidelines))

I'll fully admit I don't have plans for this, or a comprehensive plan to moderate it beyond myself. If y'all start using this we'll figure that all out.

  To get started, feel free comment below with an introduction, and what you'd like to do with this kit, or with the NES in general!

**Comment by pagehm1** [comment]
### Adding constant scrolling to the Game?

Hello, first, I want to say I love this codebase and what is has to offer. It's helped me transition a lot easier into NES game development!
That being said, there was one feature I was curious about, the map updating. I think it would be cool to have the map update AS the player moves, like Final Fantasy, rather than waiting until the player gets to the edge of the screen. Looking at the code, it would seem to be a very involved change. I was wondering, @potatolain , have you considered this as another option? Would it be possible for someone like me to develop this as a feature? Thanks.

**Reply by potatolain** [reply]
Hey there!

I have actually considered it before, but as you found it looked very involved, and I was afraid I would not be able to write something that another person could easily understand. That said, if you're willing to try, I'd be willing to support you! 

I'd ask you to plan for it to be a separate branch, like is done in a few chapters in section 3. Try to document things well too, then get it in a pull request and we can figure out a path forward. If you have the energy/writing focus, getting the start of a new chapter on how to adopt it would also be welcome. 

If you hit snags, especiallywith how the existing engine works, reach out! Full disclosure I will be away on travel from tomorrow through early next week, so I may be less responsive, but I'm definitely interested in working on this with you!

**Reply by potatolain** [reply]
I wrote that assuming you meant as a feature of the engine, but if you just want it for your own game, I'll support you in that too :)

**Reply by pagehm1** [reply]
Thanks for the quick reply! This is definitely something I would be interested in developing overtime. I haven't been involved in nes development long, so it would definitely take me a while to get accustomed to this sort of development, as well as your codebase. But I appreciate the potential support for this, and once I am more confident in creating something usable, I will get back to you. Thank you for your time, hope to talk again soon!
